src/Sarif.Driver/Sdk/MultithreadedAnalyzeCommandBase.cs

```diff
@@ -28,7 +28,7 @@ public abstract class MultithreadedAnalyzeCommandBase<TContext, TOptions> : Plug
         internal bool _captureConsoleOutput;
 
         internal ConsoleLogger _consoleLogger;
-        internal ConcurrentDictionary<string, IAnalysisLogger> _analysisLoggerCache;
+        internal ConcurrentDictionary<string, CachingLogger> _analysisLoggerCache;
```

Comment:
>ConcurrentDictionary [](http://example.com/codeflow?start=17&length=20)

This data structure *MUST* only operate on caching logger instances and so I've made this concrete in the type.